src/odemis/gui/comp/overlay/base.py

```diff
@@ -1109,42 +1135,43 @@ def __init__(self, colour=gui.SELECTION_COLOUR, center=(0, 0)):
 
         DragMixin.__init__(self)
 
-        self.v_point1 = None
-        self.v_point2 = None
-        self.v_point3 = None
-        self.v_point4 = None
+        self.v_point1: Vec = None
+        self.v_point2: Vec = None
+        self.v_point3: Vec = None
+        self.v_point4: Vec = None
 
         self.edit_v_start_pos = None  # The view port coordinates where a drag/edit originated
         self.edit_hover = None  # What edge is being edited (gui.HOVER_*)
         self.edit_v_point_idx = None
 
         self.hover = gui.HOVER_NONE
-        self.hover_direction = "NS"  # "NS" or "WE"
+        self.hover_direction = gui.HOVER_DIRECTION_NS  # "NS" or "WE"
 
         # Selection modes (none, create, edit and drag)
         self.selection_mode = SEL_MODE_NONE
 
         # This attribute can be used to see if the canvas has shifted or scaled
         self._last_shiftscale = None
 
-        self.v_edges = {}
+        # Dict in which key is the gui.HOVER_* type and value is a list of EdgeBoundingBox
+        # associated to the hover type
+        self.v_edges: Dict[int, List[EdgeBoundingBox]] = {}
 
         # TODO: Move these to the super classes
         self.colour = conversion.hex_to_frgba(colour)
         self.highlight = conversion.hex_to_frgba(gui.FG_COLOUR_HIGHLIGHT)
         self.center = Vec(center)
```

Comment:
Yeah the `.center` and `.rotation` attributes are used by external classes specifically during drawing the rotation label

Reply:
Which class is that? Isn't it some inheriting classes? As long as these are classes that inherit from the Mixin, it would be fine to mark these as private. The opposite would be to leave them "public" so that the rest of the GUI can depend on these attributes to know the center and rotation.

Reply:
> The opposite would be to leave them "public" so that the rest of the GUI can depend on these attributes to know the center and rotation.
When I start working on the Project Manager, the center and rotation will mostly become a VA. Users can input the values and the shape will be shifted or rotated accordingly.